### PR TITLE
PACA-232: Add hashed MBI to MDC for inclusion in access log

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-232](https://jira.cms.gov/browse/PACA-232)

**User Story or Bug Summary:**

Ensure that MBI Hash and JobID are logged in access logs for pre-adjudicated claims.

---

### What Does This PR Do?

Adds `mbi_hash` property to MDC when a Claim or ClaimResponse request is received with a hashed MBI as a parameter.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:
* Is this likely to address the goals expressed in the user story?
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProvider.java [hunk 1]
```diff
@@ -249,6 +250,9 @@ public Bundle findByPatient(
                 getResourceTypes(), mbiString, isHashed, excludeSamhsa, lastUpdated, serviceDate);
       }
 
+      if (isHashed) {
+        TransformerUtils.logMbiHashToMdc(mbiString);
+      }
```
Comment (KyRobbins) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProvider.java:255]: Should we move this up to the top, just after the `isHashed` computation, incase any logging logic is mistakenly added before it?

Comment (brianburton) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProvider.java:255]: Sounds good.

**Review by KyRobbins — APPROVED** [review]

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java [hunk 2]
```diff
@@ -3264,6 +3264,12 @@ public static void logBeneIdToMdc(Collection<String> beneIds) {
     }
   }
 
+  public static void logMbiHashToMdc(String mbiHash) {
```
Comment (ericbuckley) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:3267]: We need to move this to https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java#L1603

Comment (brianburton) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java]: I have moved the method.

**Review by KyRobbins — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Comment by christrotteradhoc** [comment]
Looks fine to me. Any concerns on how those logs become less useful when MBIs are re-hashed?

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**